spec/functional/x_cache_headers_spec.rb

```diff
@@ -43,18 +68,44 @@
     expect(DOMAIN + '/correct-true').to be_served_from_origin('originsite.example.com')
   end
 
+  it 'should succeed when the response with 200 status and correct origin in x-true-cache-key' do
+    net_http_res = double('response',
+                          :to_hash => {
+                              "Status" => ["200 OK"],
+                              'x-cache-key' => ['A/B/1234/123456/000/originsite.example.com/'],
+                              'x-true-cache-key' => ['A/B/1234/123456/000/originsite.example.com/']
+                          },
+                          :code => 200)
+    request = double('http request', :user => nil, :password => nil, :url => "")
+    response = RestClient::Response.create({}, net_http_res, {}, request)
+    expect(response).to be_served_from_origin('originsite.example.com')
+  end
+
+  it 'should fail when the response without x-true-cache-key and x-cache-key ' do
+    net_http_res = double('response',
+                          :to_hash => {
+                              "Status" => ["200 OK"]
+                          },
+                          :code => 200)
+    request = double('http request', :user => nil, :password => nil, :url => "")
+    response = RestClient::Response.create({}, net_http_res, {}, request)
+    expect { expect(response).to be_served_from_origin('originsite.example.com') }
+        .to raise_error (RSpec::Expectations::ExpectationNotMetError)
+  end
```

Comment:
Thanks for the extra tests! Could you please also add one for when the response contains the incorrect origin?